cmd/system-probe/utils/memory_monitor_linux.go

```diff
@@ -70,17 +70,17 @@ func getActionCallback(action string) (func(), string, error) {
 				return
 			}
 
-			defer memProfile.Close()
+			defer func() {
+				if err := memProfile.Close(); err != nil {
+					log.Errorf("Failed to generate memory profile: %s", err)
```

Comment:
This will not affect the function return value, is that intended?

Reply:
Indeed, since `getActionCallback` returns a function with no return value, I guess the only way to handle an error on `memProfile.Close()` is to log the error.

Reply:
Ah, yes, since this is the within the callback function.